src/libcmd/repl.cc

```diff
@@ -93,9 +93,16 @@ struct NixRepl
     void evalString(std::string s, Value & v);
     void loadDebugTraceEnv(DebugTrace & dt);
 
-    typedef std::set<Value *> ValuesSeen;
-    std::ostream & printValue(std::ostream & str, Value & v, unsigned int maxDepth);
-    std::ostream & printValue(std::ostream & str, Value & v, unsigned int maxDepth, ValuesSeen & seen);
+    void printValue(std::ostream & str,
+                              Value & v,
+                              unsigned int maxDepth = std::numeric_limits<unsigned int>::max())
+    {
+        ::nix::printValue(*state, str, v, PrintOptions {
```

Comment:
Not sure why but this wouldn't resolve until I qualified it with `::nix::`.

Reply:
Probably because the members of `struct NixRepl` are in scope.
Apparently C++ doesn't apply its overloading logic, but assumes that you refer to the `NixRepl` method.
Anyway, this very explicit syntax makes it very clear which `printValue` gets called, so this is fine.